### PR TITLE
[torch] Support `torch.aten.view.dtype` conversion to `flow`

### PR DESCRIPTION
**Pull request by rsuderman**
Convert to `flow.tensorbitcast` for supporting `i4` operations.

### CODE REVIEW

File: compiler/plugins/input/Torch/InputConversion/BitCastQuantTensor.cpp [hunk 1]
```diff
@@ -20,6 +21,47 @@ namespace mlir::iree_compiler::TorchInput {
 
 namespace {
 
+class BitCastViewDtypeMatmul
```
Comment (stellaraccident) [inline compiler/plugins/input/Torch/InputConversion/BitCastQuantTensor.cpp]: Copy-pasta? Doesn't seem to have anything to do with matmul.

**Review by stellaraccident — APPROVED** [review]
Test?

File: compiler/plugins/input/Torch/InputConversion/BitCastQuantTensor.cpp (hunk 1 above)
Comment (rsuderman) [reply compiler/plugins/input/Torch/InputConversion/BitCastQuantTensor.cpp]: Fixed.

**Comment by rsuderman** [comment]
> Test?

Added

**Review by stellaraccident — APPROVED** [review]

**Pull request merged**